src/main/java/mods/railcraft/common/blocks/tracks/TrackSwitchBase.java

```diff
@@ -27,6 +35,13 @@
     protected boolean switched;
     private byte sprung;
     private byte locked;
+    protected Set<EntityMinecart> lockingCarts = new HashSet<EntityMinecart>();
+    protected Set<EntityMinecart> springingCarts = new HashSet<EntityMinecart>();
```

Comment:
Why not just cache the UUID? Wouldn't that negate the need for finding the entity?